R/mediate.R

```diff
@@ -0,0 +1,394 @@
+#'   \item{Mediator}{the mediator variable}
+#'   \item{Outcome}{the outcome variable}
+#'   \item{ACME_estimate}{the Average Causal Mediation Effect (ACME) estimate}
+#'   \item{ADE_estimate}{the Average Direct Effect (ADE) estimate}
+#'   \item{ACME_pval}{the p-value for the ACME estimate}
+#'   \item{ADE_pval}{the p-value for the ADE estimate}
+#'   \item{ACME_adjpval}{the adjusted p-value for the ACME estimate}
+#'   \item{ADE_adjpval}{the adjusted p-value for the ADE estimate}
+#'   \item{ACME_CI_lower}{the lower limit of the confidence interval for the ACME estimate}
+#'   \item{ACME_CI_upper}{the upper limit of the confidence interval for the ACME estimate}
+#'   \item{ADE_CI_lower}{the lower limit of the confidence interval for the ADE estimate}
+#'   \item{ADE_CI_upper}{the upper limit of the confidence interval for the ADE estimate}
+#' }
+#'
+#' @name getMediation
+#'
+#' @examples
+#' # Import libraries
+#' library(mia)
+#' library(microbiomeDataSets)
+#' library(scater)
+#' 
+#' # Load dataset
+#' tse <- LahtiWAData()
+#'  
+#' # Convert BMI variable to numeric
+#' tse$bmi_group <- as.numeric(tse$bmi_group)
+#' 
+#' # Analyse mediated effect of nationality on BMI through alpha diversity 
+#' med_df <- getMediation(tse,
+#'                        outcome = "bmi_group",
+#'                        treatment = "nationality",
+#'                        mediator = "diversity",
+#'                        covariates = c("sex", "age"),
+#'                        treat.value = "Scandinavia",
+#'                        control.value = "CentralEurope",
+#'                        boot = TRUE, sims = 1000)
+#' 
+#' # Apply clr transformation to counts assay
+#' tse <- transformAssay(tse,
+#'                       method = "clr",
+#'                       pseudocount = 1)
+#'
+#' # Analyse mediated effect of nationality on BMI through clr-transformed features     
+#' med_df <- getMediation(tse,
+#'                        outcome = "bmi_group",
+#'                        treatment = "nationality",
+#'                        assay.type = "clr",
+#'                        covariates = c("sex", "age"),
+#'                        treat.value = "Scandinavia",
+#'                        control.value = "CentralEurope",
+#'                        boot = TRUE, sims = 300)
+#'  
+#' # Perform ordination
+#' tse <- runMDS(tse, name = "MDS",
+#'               method = "euclidean",
+#'               assay.type = "clr",
+#'               ncomponents = 3)
+#' 
+#' # Analyse mediated effect of nationality on BMI through NMDS components        
+#' med_df <- getMediation(tse,
+#'                        outcome = "bmi_group",
+#'                        treatment = "nationality",
+#'                        dim.type = "MDS",
+#'                        covariates = c("sex", "age"),
+#'                        treat.value = "Scandinavia",
+#'                        control.value = "CentralEurope",
+#'                        boot = TRUE, sims = 500)
```

Comment:
I think that the the "sims" argument is "niter" or "n.iter" in many packages. We should aim to harmonize argument names. I would go with n.iter - can we change? However if this comes directly from the mediation package, then we should keep the same and just pass the argument.

Reply:
sims is an argument of the original mediate function, so we can keep it